avaframe/com1DFA/com1DFATools.py

```diff
@@ -380,3 +382,63 @@ def checkCfgInfoType(cfgInfo):
         raise AssertionError(message)
 
     return typeCfgInfo
+
+
+def updateResCoeffFields(fields, cfg, t, dem):
+    """ update fields of cRes and detK, coefficients of resistance parameter and detrainment parameter
+        according to the thresholds of FV and FT
+        if FV OR FT below min thresholds -> apply detrainment in that area
+        if FV AND FT above min thresholds AND below max thresholds -> apply cResH (additional friction) in that area
+        if FV OR FT above max thresholds -> no impact of resistance area
+
+        Parameters
+        ------------
+        fields: dict
+            dictionary with cResRasterOrig, cResRaster and detRasterOrig, detRaster fields
+        cfg: configparser object
+            configuration of com1DFA, thresholds
+
+        Returns
+        ------------
+        fields: dict
+            updated cRes and detK fields
+    """
+
+    # fetch cRes and detK raster and thresholds for FV and FT
+    cResOrig = fields['cResRasterOrig'].copy()
+    detOrig = fields['detRasterOrig'].copy()
+    vMin = cfg.getfloat('forestVMin')
+    thMin = cfg.getfloat('forestThMin')
+    vMax = cfg.getfloat('forestVMax')
+    thMax = cfg.getfloat('forestThMax')
+
+    # create new rasters using FV, FT and thresholds to mask
+    detRaster = np.where(((fields['FV'] <= vMin) | (fields['FT'] <= thMin)), detOrig, 0.)
+    cResRasterInt = np.where(
+        ((fields['FV'] > vMin ) & ((fields['FT'] > thMin))), cResOrig,
+        0.)
+    cResRaster = np.where(((fields['FV'] > vMax) | ((fields['FT'] > thMax))), 0, cResRasterInt)
+
+    if len(np.where((detRaster > 0) & (cResRaster > 0))[0]) > 0:
+        message = 'Detrainment and increased friction within same cell!'
+        log.error(message)
+        raise AssertionError(message)
+
+    # if max thresholds are exceeded: forest destroyed remove forest
+    lTh = len(np.where((fields['FV'] > vMax) | (fields['FT'] > thMax))[0])
+    if lTh > 0:
+        cResOrig = np.where(((fields['FV'] > vMax) | (fields['FT'] > thMax)), 0, cResOrig)
+        detOrig = np.where(((fields['FV'] > vMax) | (fields['FT'] > thMax)), 0, detOrig)
+        fields['cResRasterOrig'] = cResOrig
+        fields['detRasterOrig'] = detOrig
+        log.info('Resistance area removed %d cells because FV or FT exceeded %.2f ms-1, %.2f m' % (lTh, vMax, thMax))
+        outFileRes = pathlib.Path(cfg['avalancheDir'], 'Outputs', 'com1DFA', 'reports',
+                                  ('resArea_t%.2f' % t))
+        IOf.writeResultToRaster(dem["originalHeader"], fields['cResRasterOrig'], outFileRes, flip=True)
+
+    # update fields dictionary
+    fields['cResRaster'] = cResRaster
+    fields['detRaster'] = detRaster
```

Comment:
These end up in the output, right? Then please add them to https://docs.avaframe.org/en/latest/moduleCom1DFA.html#output

Reply:
not to be exported it is just to store them during the computation so that they can be used during the computation